src/js/node/http.ts

```diff
@@ -149,6 +151,13 @@ var FakeSocket = class Socket extends Duplex {
     return this;
   }
 
+  end() {
+    if (socketMap.has(this.#id)) {
```

Comment:
We can skip the `.has` call, `delete` returns true if the key matched
```suggestion
    if (socketMap.delete(this.#id)) {
```